### PR TITLE
MacOS support

### PR DESCRIPTION
**Pull request by sallaben**
Adds MacOS >=10.15 support to `flutter_web_auth`.

Please let me know if you have any questions or suggestions!

### CODE REVIEW

**Comment by sallaben** [comment]
I would suggest browsing the difference by commit. 

cfceb4d has the class itself
0e0c773 has /example/macos files
26180cb has GIF of macos behavior.

**Comment by sallaben** [comment]
@LinusU would like to use this in my project and I'm sure others would too :)

**Comment by LinusU** [comment]
Sorry for the delay, will try to take a look this weekend 👀

**Comment by sallaben** [comment]
Hey, any luck?

File: pubspec.yaml [hunk 1]
```diff
@@ -25,8 +25,14 @@ flutter:
   # be modified. They are used by the tooling to maintain consistency when
   # adding or updating assets for this project.
   plugin:
-    androidPackage: com.linusu.flutter_web_auth
-    pluginClass: FlutterWebAuthPlugin
+    platforms:
```
Comment (LinusU) [inline pubspec.yaml:28]: This breaks older versions of Flutter so I will release this as a breaking change

**Comment by LinusU** [comment]
> (i promise this is my last bump tho)

Don't feel bad bumping! I have 321 unread notifications that I still have to get to right now, and many of them are just noise, so this puts it at the top 😅 

I'm pushing some fixups directly to your branch, for future reference please don't bump the version in the PR, I want to do that in a separate commit. (in fact, I'm automating it with [emoji-commit](https://github.com/LinusU/emoji-commit))

Also I would recommend to configure your editor to automatically insert final newline character to avoid this:

Screenshot 2020-04-13 at 10 26 01

and to remove trailing whitespace to avoid this:

Screenshot 2020-04-13 at 10 27 24

I also ran ImageOptim on the images to save some space, trimmed a few frames from the gif, added macOS info to the readme and some other minor cleanups.

**Comment by LinusU** [comment]
Thank you so much for this 🎉

**Comment by LinusU** [comment]
Released as [🚢 0.2.0 / 2020-04-13](https://github.com/LinusU/flutter_web_auth/releases/tag/0.2.0)

**Comment by eldadfux** [comment]
Great work!! @LinusU @sallaben 👏🏻

**Pull request merged**